src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.InlineContext.cs

```diff
@@ -0,0 +1,582 @@
+                SyntaxNode statementContainsCallee,
+                MethodParametersInfo methodParametersInfo,
+                CancellationToken cancellationToken)
+            {
+                var syntaxFacts = inlineMethodRefactoringProvider._syntaxFacts;
+                var semanticFactsService = inlineMethodRefactoringProvider._semanticFactsService;
+                var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var syntaxGenerator = SyntaxGenerator.GetGenerator(document);
+                var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var inlineNode = inlineMethodRefactoringProvider.GetInlineStatement(calleeMethodDeclarationNode);
+
+                // Get the parameterSymbol to identifier's name mappings.
+                // Example:
+                // void Caller()
+                // {
+                //     int i = 10;
+                //     Callee(i);
+                // }
+                // void Callee(int a)
+                // {
+                //     DoSomething(a);
+                // }
+                // Input is an array of (ParameterSymbol(int a), IdentifierSyntaxNode(int i))
+                // Output is an array of (ParameterSymbol(int a), string i)
+                var parametersWithIdentifierArgumentToName = methodParametersInfo.ParametersWithIdentifierArgument
+                    .Select(parameterAndArgument =>
+                        (parameterAndArgument.parameterSymbol, semanticModel.GetSymbolInfo(parameterAndArgument.identifierSyntaxNode, cancellationToken).GetAnySymbol()?.Name))
+                    .Where(parameterAndName => parameterAndName.Name != null)
+                    .ToImmutableArray();
+
+                // Get the parameterSymbol to variable declaration name mapping.
+                // Example:
+                // void Caller()
+                // {
+                //     Callee(out int i);
+                // }
+                // void Callee(out int a)
+                // {
+                //     DoSomething(out a);
+                // }
+                // Input is an array of (ParameterSymbol(int a), IdentifierSyntaxNode(int i))
+                // Output is an array of (ParameterSymbol(int a), string i)
+                var parametersWithVariableDeclarationArgumentToName = methodParametersInfo
+                    .ParametersWithVariableDeclarationArgument
+                    .Select(parameterAndArgument =>
+                        (parameterAndArgument.parameterSymbol,
+                            semanticModel
+                                .GetSymbolInfo(parameterAndArgument.variableDeclarationNode, cancellationToken)
+                                .GetAnySymbol()?.Name))
+                    .Where(parameterAndName => parameterAndName.Name != null)
+                    .ToImmutableArray();
```

Comment:
curious why thse are arrays rather that dictionaries.  it seems like it's always a mapping from param->name, so a dict seems better.  but maybe there's a good reason for arrays here?

Reply:
I remembered that in one version I have tried to use dictionary in all these places. But since the element becomes unordered then I could get some thing like this 
Before:
`void Caller
{
Callee(Foo(), Bar())
}`
`void Callee(int a, int b) {}`
After:
`void Caller() 
{
     int b = bar();
    int a = Foo();
     Callee()
}`
`void Callee(int a, int b) {}`
Notice that variable b is put in font of variable a. 
For storing identifier it won't be a problem, but I just choose to use array all the place to make it clearly. ( I feel it doesn't have any perf/mem should be the same)